src/modules/workspaces/workspaces.controller.spec.ts

```diff
@@ -240,5 +240,73 @@ describe('Workspace Controller', () => {
       );
       expect(getMembers).toEqual(mockResolvedValues);
     });
+
+    describe('GET /invitations/:inviteId/validate', () => {
+      const mockResponse = {
+        status: jest.fn().mockReturnThis(),
+        send: jest.fn().mockReturnThis(),
+      };
+      it('When invitation is validated successfully, then it returns.', async () => {
+        const invite = newWorkspaceInvite();
+        workspacesUsecases.validateWorkspaceInvite.mockResolvedValueOnce(
+          invite.id,
+        );
+
+        const validateInvite =
+          await workspacesController.validateWorkspaceInvitation(
+            invite.id,
+            mockResponse as any,
+          );
+
+        expect(workspacesUsecases.validateWorkspaceInvite).toHaveBeenCalledWith(
+          invite.id,
+        );
+        expect(validateInvite).toEqual(invite.id);
+      });
+
+      it('When invitation is validated with invalid inviteId, then it should throw BadRequestException', async () => {
+        const errorMessage = 'Invalid invitation ID';
+        workspacesUsecases.validateWorkspaceInvite.mockRejectedValueOnce(
+          new BadRequestException(errorMessage),
+        );
+
+        try {
+          await workspacesController.validateWorkspaceInvitation(
+            'invalid-id',
+            mockResponse as any,
+          );
+          // If the above line does not throw, the test should fail.
+          fail(
+            'Expected validateWorkspaceInvitation to throw a BadRequestException.',
+          );
+        } catch (error) {
+          // Now we assert that the error is what we expect it to be.
+          expect(error).toBeInstanceOf(BadRequestException);
+          expect(error.response).toEqual({
+            statusCode: 400,
+            message: errorMessage,
+            error: 'Bad Request',
+          });
```

Comment:
We do not really need to test the thrown text. You can follow the previous tests and do it  this way

```javascript
await expect(workspacesController.validateWorkspaceInvitation(
            'invalid-id',
            mockResponse)).rejects.toThrow(BadRequestException);
```